app/models/ability.rb

```diff
@@ -55,6 +56,35 @@ def initialize(member)
       end
       # note we don't support update for notifications
 
+      # can read seed requests if you are requester or seed owner
+      can :read,   SeedTrade, requester_id: member.id
+      can :read,   SeedTrade, seed: { :owner_id => member.id }
+      can :create, SeedTrade, requester_id: member.id
+
+      # you can't create a request for yourself
+      cannot :create, SeedTrade, seed: { :owner_id => member.id }
+
+      # you can decline if the trade was not accepted or declined
```

Comment:
Well, I did not considered this point of view. When I built it, I thought the javascript alert asking "are you sure?" would be enough.

what approach would you use here, stop restrictions on those buttons?